### PR TITLE
Fix some -Wfloat-conversion warnings (Pt. 8)

### PR DESCRIPTION
**Pull request by Holzhaus**
Fixes the remaining `-Wfloat-conversion` warnings and cleans up the `WaveformRendererPreroll` drawing code as requested by @daschuer here: https://github.com/mixxxdj/mixxx/pull/3129#discussion_r506197742

### CODE REVIEW

**Comment by Holzhaus** [comment]
CI failures are unrelated.

**Comment by uklotzde** [comment]
Fixes https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/100.25.20CPU.20after.20track.20ended

File: src/waveform/renderers/waveformrendererpreroll.cpp [hunk 1]
```diff
@@ -40,20 +40,21 @@ void WaveformRendererPreroll::draw(QPainter* painter, QPaintEvent* event) {
...
-        //         << "totalSamples" << totalSamples
-        //         << "WaveformRendererPreroll::playMarkerPosition=" << playMarkerPosition;
-
-        const int polyLength = static_cast<int>(40.0 / samplesPerPixel);
+        const double playMarkerPositionFrac = m_waveformRenderer->getPlayMarkerPosition();
+        const double vSamplesPerPixel = m_waveformRenderer->getVisualSamplePerPixel();
+        const double numberOfVSamples = m_waveformRenderer->getLength() * vSamplesPerPixel;
+
+        int currentVSamplePosition = m_waveformRenderer->getPlayPosVSample();
+        int totalVSamples = m_waveformRenderer->getTotalVSample();
```
Comment (uklotzde) [inline src/waveform/renderers/waveformrendererpreroll.cpp:48]: const (both)

File: src/waveform/renderers/waveformrendererpreroll.cpp [hunk 2]
```diff
@@ -40,20 +40,21 @@ void WaveformRendererPreroll::draw(QPainter* painter, QPaintEvent* event) {
...
+        // qDebug() << "currentVSamplePosition" << currentVSamplePosition
+        //          << "lastDisplayedPosition" << lastDisplayedPosition
+        //          << "vSamplesPerPixel" << vSamplesPerPixel
+        //          << "numberOfVSamples" << numberOfVSamples
+        //          << "totalVSamples" << totalVSamples
+        //          << "WaveformRendererPreroll::playMarkerPosition=" << playMarkerPositionFrac;
+
+        const double polyLength = 40.0 / vSamplesPerPixel;
+        const double polyVSampleSize = (polyLength + 1) * vSamplesPerPixel;
```
Comment (uklotzde) [inline src/waveform/renderers/waveformrendererpreroll.cpp:57]: The expression `polyLength + 1` appears multiple times in the following code. Can we assign it to a constant with a meaningful name that also communicates the *plus one* rationale?

**Review by uklotzde — COMMENTED** [review]
Works as expected. @daschuer?

File: src/waveform/renderers/waveformrendererpreroll.cpp (hunk 2 above)
Comment (Holzhaus) [reply src/waveform/renderers/waveformrendererpreroll.cpp]: Done. I'm not 100% sure but think the `+ 1` is used for the offset to accomodate the line width of the QPainter. I changed it to `painter->pen().lineWidthF()` to make this explicit.

**Review by uklotzde — APPROVED** [review]

**Comment by uklotzde** [comment]
Merging this now because Mixxx 2.3 beat is unusable without this fix. LGTM

**Comment by uklotzde** [comment]
Retargeted to 2.3!

**Comment by Holzhaus** [comment]
> Retargeted to 2.3!

Oops, that was the plan all along. 

> Merging this now because Mixxx 2.3 beta is unusable without this fix. LGTM

Alright. Please believe me that I didn't break 2.3 intentionally to get my PRs merged faster :D

**Pull request merged**